crates/ditto-ast/src/expression.rs

```diff
@@ -173,6 +189,8 @@ impl Expression {
                     return_type: Box::new(body.get_type()),
                 }
             }
+            // Maybe this is correct???
+            Self::If { true_clause, .. } => true_clause.get_type(),
```

Comment:
This is totally fine, but I have a slight preference for storing the type in a field during type checking. I do a similar thing for `call_type` (see above).